policy/release/test.rego

```diff
@@ -35,28 +36,23 @@ deny[result] {
 # METADATA
 # title: Some tests did not pass
 # description: |-
-#   Enterprise Contract requires that all the tests in the
-#   test results have a result of 'SUCCESS'. This will fail if any
-#   of the tests failed and the failure message will list the names
-#   of the failing tests.
+#   Enterprise Contract requires that all the tests in the test results
+#   have a successful result. A successful result is one that isn't a
+#   "FAILURE" or "ERROR". This will fail if any of the tests failed and
+#   the failure message will list the names of the failing tests.
 # custom:
 #   short_name: test_result_failures
 #   failure_msg: "The following tests did not complete successfully: %s"
 #
 deny[result] {
-	# Collect all failed tests and convert their name to "test:<name>" format
-	all_failed := {failure |
-		result := lib.results_from_tests[_]
-		result.result != "SUCCESS"
-		failure := sprintf("test:%s", [result.__task_name])
-	}
+	all_failed = results_with_status({"FAILURE", "ERROR"})
```

Comment:
If there is a typo on the status, say "EROR", I think the test be marked as being successful. Previously, a typo would assume a failure which I think is a safer  default. Maybe we need to add a check to ensure the state is one of the known states?

Reply:
Perhaps as a warning and in another pull request.